src/Console/CommandRunner.php

```diff
@@ -157,11 +161,15 @@ protected function getShell(ConsoleIo $io, CommandCollection $commands, $name)
                 " Run `{$this->root} --help` to get the list of valid commands."
             );
         }
-        $classOrInstance = $commands->get($name);
-        if (is_string($classOrInstance)) {
-            return new $classOrInstance($io);
+        $instance = $commands->get($name);
+        if (is_string($instance)) {
+            $instance = new $instance($io);
+        }
+        // TODO Should this be an interface?
+        if (method_exists($instance, 'setCommandCollection')) {
```

Comment:
I'm contemplating doing this at the same time as propagating the root command name. Right now we have a few places we hard code `cake` when it should really reflect the root command name given to the `CommandRunner`.